components/post_view/post_body/post_body.tsx

```diff
@@ -17,89 +20,93 @@ import PostBodyAdditionalContent from 'components/post_view/post_body_additional
 import PostMessageView from 'components/post_view/post_message_view';
 import ReactionList from 'components/post_view/reaction_list';
 import LoadingSpinner from 'components/widgets/loading/loading_spinner';
+import {PluginsState} from 'types/store/plugins';
 
 const SENDING_ANIMATION_DELAY = 3000;
 
-export default class PostBody extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * The post to render the body of
-         */
-        post: PropTypes.object.isRequired,
-
-        /**
-         * The parent post of the thread this post is in
-         */
-        parentPost: PropTypes.object,
-
-        /**
-         * The poster of the parent post, if exists
-         */
-        parentPostUser: PropTypes.object,
-
-        /**
-         * The function called when the comment icon is clicked
-         */
-        handleCommentClick: PropTypes.func.isRequired,
-
-        /**
-         * Set to render post body compactly
-         */
-        compactDisplay: PropTypes.bool,
-
-        /**
-         * Set to highlight comment as a mention
-         */
-        isCommentMention: PropTypes.bool,
-
-        /**
-         * Set to render a preview of the parent post above this reply
-         */
-        isFirstReply: PropTypes.bool,
-
-        /*
-         * Post type components from plugins
-         */
-        pluginPostTypes: PropTypes.object,
-
-        /**
-         * Flag passed down to PostBodyAdditionalContent for determining if post embed is visible
-         */
-        isEmbedVisible: PropTypes.bool,
-
-        /**
-         * Whether or not the post username can be overridden.
-         */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
-
-        /**
-         * Set not to allow edits on post
-         */
-        isReadOnly: PropTypes.bool,
-    }
-
+type Props = {
+
+    /**
+   * The post to render the body of
+   */
+    post: Post;
+
+    parentPost?: Post;
+
+    /**
+   * The parent post of the thread this post is in
+   */
+    parentPostUser?: UserProfile | null;
+
+    /**
+   * Set to render a preview of the parent post above this reply
+   */
+    isFirstReply?: boolean;
+
+    /**
+   * The function called when the comment icon is clicked
+   */
+    handleCommentClick: () => void;
+
+    /**
+   * Set to render post body compactly
+   */
+    compactDisplay?: boolean;
+
+    /**
+   * Set to highlight comment as a mention
+   */
+    isCommentMention?: boolean;
+
+    /**
+   * Post type components from plugins
+   */
+    pluginPostTypes?: PluginsState['postTypes'];
+
+    /**
+   * Flag passed down to PostBodyAdditionalContent for determining if post embed is visible
+   */
+    isEmbedVisible?: boolean;
+
+    /**
+   * Whether or not the post username can be overridden.
+   */
+    enablePostUsernameOverride?: boolean;
```

Comment:
This is required in the original component.

Reply:
Which component are you referring to ? @devinbinnie

Reply:
In `post_body.jsx`:

```
        /**
         * Whether or not the post username can be overridden.
         */
        enablePostUsernameOverride: PropTypes.bool.isRequired,
```

Reply:
I guess we can not add Proptypes directly in .ts file. Can I make `enablePostUsernameOverride ?: boolean` to `enablePostUsernameOverride : boolean`?

Reply:
Yes that should be correct.